cmd/buildah/manifest.go

```diff
@@ -328,13 +328,22 @@ func manifestAddCmd(c *cobra.Command, args []string, opts manifestAddOpts) error
 	digest, err := list.Add(getContext(), systemContext, ref, opts.all)
 	if err != nil {
 		var storeErr error
-		// check if the local image exists
-		if ref, _, storeErr = util.FindImage(store, "", systemContext, imageSpec); storeErr != nil {
-			return err
+		// Add operation for list failed with error as an alternative
+		// we are looking for image in local storage by name,
+		// at a given point of time there will be only one image
+		// mapped by name so don't set systemContext otherwise
+		// lookup will fail if user wants to set custom arch or
+		// variant for this image. Instead perform a generic lookup
+		// on host since `buildah manifest add --variant <> list` or
+		// `buildah manifest add --arch <> list` expects to override
+		// arch or variant and not perform a lookup by that.
+		// See: https://github.com/containers/buildah/issues/3511
```

Comment:
```suggestion
		// Retry without a custom system context.  A user may want to add
		// a custom platform (see #3511).
```
I find the comment too verbose.